rocketpool-daemon/common/validator/key-recovery-manager/key-recovery-manager.go

```diff
@@ -0,0 +1,14 @@
+package key_recovery_manager
+
+import (
+	"github.com/rocket-pool/node-manager-core/beacon"
+)
+
+type KeyRecoveryManager interface {
+	RecoverMinipoolKeys() ([]beacon.ValidatorPubkey, map[beacon.ValidatorPubkey]error, error)
```

Comment:
can this just be argumented with `enablePartialRecovery bool` instead of having two different implementations? Might reduce repeated code.

Reply:
Discussed in discord. Would like to avoid parameterizing the condition. Will explore multiple methods approach.